frontend/src/components/UserForms/LocationsForm.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import { Location } from '../../types/index';
+
+type FormProps = {
+  onClick: ((newLocation: Location) => void);
+}
+
+const Form = ({ onClick }: FormProps) => {
+  const [newLocation, setnewLocation] = useState<Location>({
+    id: '',
+    name: '',
+    address: '',
+  });
+  const [validName, setValidName] = useState(false);
+  const [validAddress, setValidAddress] = useState(false);
+
+  const handleInput = (evt: any) => {
+    const fieldName = evt.target.name;
+    const fieldValue = evt.target.value;
+    if (fieldName === 'name') {
+      newLocation.name = fieldValue;
+      setValidName(fieldValue.length > 0);
+    } else if (fieldName === 'address') {
+      newLocation.address = fieldValue;
+      setValidAddress(fieldValue.length > 0);
+    }
+    setnewLocation(newLocation);
```

Comment:
Might be better to create a copy of newLocation and edit the values in the copy. Then at the end, you can set the newLocation to the altered copy.